src/bindings/answer.rs

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use std::{ffi::c_char, ptr::null_mut};
+
+use super::{
+    common::{borrow, borrow_mut, free, release, release_optional, release_string, string_free, string_view},
+    concept::ConceptIterator,
+    iterator::CIterator,
+    query::ConceptMapIterator,
+};
+use crate::{
+    answer::{ConceptMap, ConceptMapGroup, Explainable, Explainables, Numeric, NumericGroup},
+    common::{box_stream, stream::BoxStream},
+    concept::Concept,
+    logic::{Explanation, Rule},
+};
+
+#[no_mangle]
+pub extern "C" fn concept_map_drop(concept_map: *mut ConceptMap) {
+    free(concept_map);
+}
+
+type StringIteratorInner = CIterator<String, BoxStream<'static, String>>;
```

Comment:
I think these rather basic compound structures such as StringPair, StringIterator can go into `common.rs` since there may be places that will use them later for other APIs